src/main/java/org/folio/circulation/domain/representations/AccountRepresentation.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.circulation.domain.representations;
+
+import java.util.UUID;
+
+import org.folio.circulation.domain.FeeFine;
+import org.folio.circulation.domain.FeeFineOwner;
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.support.JsonPropertyWriter;
+
+import io.vertx.core.json.JsonObject;
+
+public class AccountRepresentation extends JsonObject {
```

Comment:
How was it decided that the representation be a specialisation of JsonObject rather than a class that converts a domain class to JSON?

Reply:
I wasn't necessarily asking for this to be changed, I wanted to understand why this was way was chosen?

Reply:
For simplicity I guess, because accountsStorageClient.post() takes JsonObject. But my understanding is that we're now trying to avoid using JSONs in the inner logic, converting them to and from the domain objects on their way in and out of the circulation app, is that correct? So I removed this JSON representation and extended Account to include all the necessary fields as well as from() and toJson() methods.

Reply:
> For simplicity I guess, because accountsStorageClient.post() takes JsonObject. 

I think it is a good idea to create a representation of the account in that can be serialized to JSON, and I'm happy for that to be a separate class.

I'd be happy with reinstating that approach, maybe calling it AccountStorageRepresentation.

Given that my question caused confusion that generated rework, this can be deferred if you want.

> But my understanding is that we're now trying to avoid using JSONs in the inner logic, converting them to and from the domain objects on their way in and out of the circulation app, is that correct? 

That understanding is correct. mod-circulation does not yet have a good common pattern for conversion to and from domain classes.

bq. So I removed this JSON representation and extended Account to include all the necessary fields as well as from() and toJson() methods.

I wasn't intending for you to have to change it to that. I think the `toJson` and `fromJson` methods are an intermediary step in trying to get the JSON out of the domain where possible.

Apologies that my question led to that assumption :-/

Reply:
No worries, I returned AccountStorageRepresentation. At least Account now has all of the account fields and no constructors accepting representation.